devmapper: Log start and end of filesystem creation

rhvgoyal opened the pull request:
Fixes #16653 

ext4 filesystem creation can take a long time on 100G thin device and
systemd might time out and kill docker service. Often user is left thinking
why docker is taking so long and logs don't give any hint. Log an info
message in journal for start and end of filesystem creation. That way
a user can look at logs and figure out that filesystem creation is
taking long time.

Signed-off-by: Vivek Goyal vgoyal@redhat.com

estesp commented on daemon/graphdriver/devmapper/deviceset.go:
```diff
@@ -590,6 +590,9 @@ func (devices *DeviceSet) createFilesystem(info *devInfo) error {
 	}
 
+	logrus.Infof("devmapper: Creating filesystem %s on device(%s). START", devices.filesystem, info.Name())
+	defer logrus.Infof("devmapper: Creating filesystem %s on device(%s). END", devices.filesystem, info.Name())
```
potentially minor, but with a `defer` executed log entry, that means that even in all error cases you get the log entry, even if no actual filesystem creation happened?  Overall this seems like a good thing to log, but wonder if it should be more clearly aligned with actual creation starting and finishing..

rhvgoyal replied:
@estesp right and that's why we never claim that filesystem creation successfully finished. We just say START and END in an attempt to figure out when does operation finish (success/failure).

If personally feel that we are already doing this in many of the functions where we are logging START/END of function and END does not mean success. It just means that you returned from function.

So I feel this is good enough. But if you feel strongly about it, I can change it.

calavera commented:
I agree with @estesp that the exit logging can be confusing. I think we should be explicit if we're going to add this information. At exit, we should say whether this operation finished successfully or not.

rhvgoyal commented:
@calavera @estesp Ok, I have pushed another version of patch which will explicitly say whether filesystem creation succeeded or not. PTAL.

calavera commented:
I know it's not part of this change, but, can we replace the 4 lines at the bottom of the function with a single `return`?

This code:

```go
 	if err != nil {
 		return err
 	}
 
 	return nil
```

Can be:

```go
    return
```

Regardless, LGTM.

rhvgoyal commented:
yep we can do that. will do it tomorrow.

rhvgoyal committed:
devmapper: Log start and end of filesystem creation

ext4 filesystem creation can take a long time on 100G thin device and
systemd might time out and kill docker service. Often user is left thinking
why docker is taking so long and logs don't give any hint. Log an info
message in journal for start and end of filesystem creation. That way
a user can look at logs and figure out that filesystem creation is
taking long time.

Signed-off-by: Vivek Goyal <vgoyal@redhat.com>

rhvgoyal commented:
@calavera made the changes as you had mentioned.

estesp commented:
LGTM
